docs/index.rst

```diff
@@ -1,260 +1,427 @@
+- A timeout in milliseconds.
+- A path to a temporary directory.
+- A log level.
 
-    config = confit.Configuration('MyGreatApp', __name__)
+Configurations are distinct from data, although the line is hard to define.
+In our approximation, an application can run without data, but it cannot run
+without a configuration. A configuration can be packaged and imported, but
+data cannot.
 
-The first parameter is required; it’s the name of your application that
-will be used to search the system for config files. The second parameter
-is optional: it’s the name of a module that will guide the search for a
-*defaults* file. Use this if you want to include a
-``config_default.yaml`` file inside your package. (The included
-``example`` package does exactly this.)
 
-Now, you can access your configuration data as if it were a simple
-structure consisting of nested dicts and lists—except that you need to
-call the method ``.get()`` on the leaf of this tree to get the result as
-a value::
+Features
+--------
 
-    value = config['foo'][2]['bar'].get()
 
-Under the hood, accessing items in your configuration tree builds up a
-*view* into your app’s configuration. Then, ``get()`` flattens this view
-into a file, performing a search through each configuration data source
-to find an answer. More on view later.
+- Put your **default configuration in code** so that your application or
+  library will run out of the box.
+- Easily let your users **override the configuration** with **command-line
+  arguments, environment variables, and configuration files in multiple
+  formats**.
+- Define your own **custom sources**, e.g., for reading configuration from a
+  database.
+- Choose a precedence order for **any number or combination of sources** that
+  make sense for you.
+- Easily **type check and validate** configuration overrides and emit **human
+  readable errors** that point users to the offending overrides.
+- Look for configuration files in **platform-specific paths** like
+  ``$XDG_CONFIG_HOME`` or ``~/.config`` on Unix; ``Application Support`` on
+  Mac OS X; and ``%APPDATA%`` on Windows.
 
-If you know that a configuration value should have a specific type, just
-pass that type to ``get()``::
 
-    int_value = config['number_of_goats'].get(int)
+Examples
+--------
 
-This way, Confit will either give you an integer or raise a
-``ConfigTypeError`` if the user has messed up the configuration. You’re
-safe to assume after this call that ``int_value`` has the right type. If
-the key doesn’t exist in any configuration file, Confit will raise a
-``NotFoundError``. Together, catching these exceptions (both subclasses
-of ``confit.ConfigError``) lets you painlessly validate the user’s
-configuration as you go.
+Hello World
+^^^^^^^^^^^
 
+::
 
-View Theory
+  1| import confit
+  2| default_config = confit.from_mapping({'greeting': 'Hello, world!'})
+  3| sys_config = confit.from_yaml('/etc/example.yml')
+  4| config = default_config + sys_config
```

Comment:
Scratch that: in this model, summation produces a new *flat* variable mapping. This way, the overriding decisions are made eagerly, rather than lazily as in the current Confit.